### ISSUE TITLE
Protected procedure call should pass immediate arguments

### ISSUE THREAD

**Issue by xurtis** [BreakawayConsulting/platform-sel4-docs#1]
A protected procedure call can pass arguments directly via the kernel in seL4 'fast message registers'. These should be use to pass function arguments in an RPC-style fashion.

**Comment by bennoleslie** [comment]
I agree. What I'm struggling with is how exactly we can implement that in a reasonable manner without and IDL compiler on day one (or developers writing things in assembler).

Suggestions welcome!
